tests/components/wmspro/test_config_flow.py

```diff
@@ -323,3 +328,87 @@ async def test_config_flow_unknown_error(
         CONF_HOST: "1.2.3.4",
     }
     assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_config_flow_duplicate_entries(
+    hass: HomeAssistant,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_test: AsyncMock,
+) -> None:
+    """Test we handle an unknown error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: "1.2.3.4",
+        },
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "1.2.3.4"
+    assert result["data"] == {
+        CONF_HOST: "1.2.3.4",
+    }
```

Comment:
This can be changed into just creating a mock config entry and adding it to hass

Reply:
Really? How would the MockConfigEntry have the runtime_data with the rooms available?

Reply:
Adding a config entry does not load it.

`hass.config_entries.async_setup(entry.entry_id)` will

Reply:
I got it working, see above regarding `await hass.async_block_till_done()`, is this fine as well? Now it is tested end-to-end.

Reply:
But we don't need to go through the config flow twice, so it would be nicer to use the mock config entry if possible

Reply:
Done, thanks for the feedback. :)